### DISCUSSION TITLE
Which composer and node versions are recommended

### DISCUSSION THREAD

**Discussion by astridx** [joomla/joomla-cms#31783] [General]
I have not found any information about recommended versions. Maybe I wasn't looking in the right place. I would have expected it here:
https://docs.joomla.org/J4.x:Setting_Up_Your_Local_Environment

While running `composer install` I get this

```
composer install

Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Nothing to install, update or remove
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
Generating optimized autoload files
Class Joomla\Tests\Unit\Libraries\Cms\Image\ImageInspector located in ./tests/Unit/Libraries/Cms/Image/stubs/ImageInspector.php does not comply with psr-4 autoloading standard. Skipping.
Class Joomla\CMS\Filesystem\Support\StringController located in ./libraries/src/Filesystem/Support/Stringcontroller.php does not comply with psr-4 autoloading standard. Skipping.
Class Joomla\CMS\Form\Rule\SubformRule located in ./libraries/src/Form/Rule/SubFormRule.php does not comply with psr-4 autoloading standard. Skipping.

                                                                                                                     
  [ErrorException]                                                                                                   
  file_put_contents(/usr/src/app/joomla-cms/libraries/vendor/bin/phpcs): Failed to open stream: No such file or dir  
  ectory                                                                                                             
           
```

`npm ci` was fine. 

## My installation:

```
composer --version
Composer version 2.0.8 2020-12-03 17:20:38
```

```
lsb_release -a
Ubuntu 20.04.1 LTS focal
```

```
node --version
v14.15.3
npm --version
6.14.10
```

**Comment by chmst** [comment]
HI, I have noet really an answer, but since the beginning, I used to update my node / composer software from time to time and never noticed any issues. So I suppose that there is no recommentation for a special version.

**Reply by astridx** [reply]
Then I ask the question differently: does someone successfully use composer version 2?

**Comment by brianteeman** [comment]
To answer part of your question
```
  "engines": {
    "node": ">=10.19",
    "npm": ">=6.13.4"
  },
```
And see 
https://github.com/joomla/joomla-cms/pull/31712 for a note about npm 7

**Comment by astridx** [comment]
Thank you @brianteeman 
Now I know that I can check this for minor versons here:  
https://github.com/joomla/joomla-cms/blob/81a5fcd2fb33a7fd01d4b184bf1a6c56348bca45/package.json#L10

I think in the case of Composer this does not support Composer 3.0 
https://github.com/joomla/joomla-cms/blob/62ad0304babd1e87a7433a928676fcee1607b433/composer.json#L62
